pkg/controller/dnsendpoint/dnsendpoint_controller.go

```diff
@@ -158,8 +158,13 @@ type ReconcileDNSEndpoint struct {
 }
 
 // Reconcile syncs the name server entries for a DNSZone subdomain in the root domain's hosted zone.
+// Discrepancies are discovered by comparing to the nameServerScraper's cache.
+// The DNSZone.Status.NameServers are considered definitive. If the scraper's cache differs, we
+// reassert the entries from the DNSZone (and update the cache to prevent thrashing).
+// If the DNSZone is marked for deletion, we delete the subdomain's records from the root domain's
+// hosted zone (and delete the cache entry).
 func (r *ReconcileDNSEndpoint) Reconcile(ctx context.Context, request reconcile.Request) (reconcile.Result, error) {
-	dnsLog := controllerutils.BuildControllerLogger(ControllerName, "dnsZone", request.NamespacedName)
+	dnsLog := controllerutils.BuildControllerLogger(ControllerName, "dnsEndpoint", request.NamespacedName)
```

Comment:
revert. The second arg describes the CRD being reconciled on. The first arg (ControllerName) is what informs the `controller=` in the logs.